lib/FilesHooks.php

```diff
@@ -231,34 +231,36 @@ protected function addNotificationsForFileAction($filePath, $activityType, $subj
 			$this->generateRemoteActivity($accessList['remotes'], $activityType, time(), $this->currentUser->getCloudId(), $accessList['ownerPath']);
 		}
 
-		$affectedUsers = $accessList['users'];
-
-		// file can be shared using GroupFolders, including ACL check
-		if ($this->config->getSystemValueBool('activity_use_cached_mountpoints', false)) {
-			$affectedUsers = array_merge($affectedUsers, $this->getAffectedUsersFromCachedMounts($fileId));
-		}
-
-		[$filteredEmailUsers, $filteredNotificationUsers] =
-			$this->getFileChangeActivitySettings($fileId, array_keys($affectedUsers));
-
-		foreach ($affectedUsers as $user => $path) {
-			$user = (string)$user;
-
-			if ($user === $this->currentUser->getUID()) {
-				$userSubject = $subject;
-				$userParams = [[$fileId => $path]];
-			} else {
-				$userSubject = $subjectBy;
-				$userParams = [[$fileId => $path], $this->currentUser->getUserIdentifier()];
+		if (!empty($accessList['users'])) {
```

Comment:
This is not correct. You are skipping the `$this->config->getSystemValueBool('activity_use_cached_mountpoints', false)` block ❗ 

The actual fix would be to:
```php
$affectedUsers = $accessList['users'] ?? [];
```
on line 234

Reply:
Godd spot. Fixed, thanks.